shm/src/item.cpp

```diff
@@ -0,0 +1,8 @@
+#include "item.hpp"
+
+Item::Item(const std::string& name, size_t amount, size_t basePrice, Rarity rarity)
+    : Cargo(name, amount, basePrice), rarity_(rarity) {}
+
+size_t Item::getPrice() const {
+    return basePrice_ * static_cast<size_t>(rarity_);

```

Comment:
Please don't do it.
Objects' properties (like 'value coefficient' of object 'rarity') should not be depending on their binary representation.
Leave enum integer values whatever compiler wants them to be.


Then store "coefficients" in some (possibly static) collection, ie. 
{ common : 1,
  rare : 2, 
  (...)
}.

I know, more code and it's slower this way - but it's the right way.

Also - we'll have to do it anyway as soon as Rarity will get second property (ie. drop rate).

Reply:
I don't know anything about the second property of Rarity. I think that I missed some part of lecture.

Reply:
You don't understand.
At some point it MIGHT get second.

Reply:
And even if it doesn't:
Why is price coefficient used as a binary representation for the enums?

Reply:
Added map with multipliers